### PR TITLE
Scrypt

### PR DESCRIPTION
**Pull request by miyconst**
Remove double hashing and send password over HTTP POST rather than query string.

Implements https://github.com/StarcounterPrefabs/SignIn/issues/34

cc @warpech, @malx122

### CODE REVIEW

File: src/SignIn/Api/MainHandlers.cs [hunk 1]
```diff
@@ -140,7 +137,25 @@ public void Register()
...
+                string rememberMe = values["rememberMe"];
+
+                HandleSignIn(username, password, rememberMe);
+                Session.Current.CalculatePatchAndPushOnWebSocket();
+
+                //Session.ScheduleTask(Session.Current.SessionId, (s, id) =>
+                //{
+                    
+                //});
+
```
Comment (warpech) [inline src/SignIn/Api/MainHandlers.cs]: Please clean up commented out code

File: src/SignIn/ViewModels/RestorePasswordFormPage.json.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ void Handle(Input.SignInClick Action)
             }
         }
 
-        void Handle(Input.Username action) // Makes the Reset Password clickable again.
+        void Handle(Input.Username Action) // Makes the Reset Password clickable again.
```
Comment (warpech) [inline src/SignIn/ViewModels/RestorePasswordFormPage.json.cs]: According to the .Net styleguide, this should be `action`.

Method parameters should be camelCase: https://msdn.microsoft.com/en-us/library/ms229043(v=vs.100).aspx

File: src/SignIn/ViewModels/SignInFormPage.json [hunk 3]
```diff
@@ -5,5 +5,6 @@
   "Message": "",
   "Submit": 0,
   "SignInClick$": 0,
-  "RestoreClick$": 0
+  "RestoreClick$": 0,
+  "Referer": ""
```
Comment (warpech) [inline src/SignIn/ViewModels/SignInFormPage.json]: Please change to `SessionUri`, as agreed in https://github.com/Starcounter/Starcounter/issues/3931

File: src/SignIn/ViewModels/SignInPage.json [hunk 4]
```diff
@@ -6,5 +6,6 @@
   "FullName": "",
   "ImageUrl": "",
   "SignInClick$": 0,
-  "Submit": 0
+  "Submit": 0,
+  "Referer": ""
```
Comment (warpech) [inline src/SignIn/ViewModels/SignInPage.json]: Please change to `SessionUri`, as agreed in https://github.com/Starcounter/Starcounter/issues/3931

File: src/SignIn/wwwroot/SignIn/elements/signin-element.html [hunk 5]
```diff
@@ -15,8 +13,9 @@
                 username: { type: String },
                 password: { type: String, notify: true },
                 rememberMe: { type: Boolean, value: false },
-                url: { type: String, value: "/signin/partial/signin/" },
-                submit: { type: Number, value: 0, observer: "submitChanged" }
+                url: { type: String, value: "/signin/partial/signin" },
+                submit: { type: Number, value: 0, observer: "submitChanged" },
+                referer: { type: String }
```
Comment (warpech) [inline src/SignIn/wwwroot/SignIn/elements/signin-element.html]: I would suggest to change this to `sessionUri` as well.

**Comment by warpech** [comment]
@miyconst In Pull Requests, please add a link to the original issue that explain what does a PR implement. I have edited OP to include that link

**Comment by miyconst** [comment]
>@miyconst In Pull Requests, please add a link to the original issue that explain what does a PR implement. I have edited OP to include that link

Okay. Thanks for the edit.

File: src/SignIn/ViewModels/SignInFormPage.json (hunk 3 above)
Comment (miyconst) [inline src/SignIn/ViewModels/SignInFormPage.json]: Done.

File: src/SignIn/ViewModels/SignInPage.json (hunk 4 above)
Comment (miyconst) [inline src/SignIn/ViewModels/SignInPage.json]: Done.

File: src/SignIn/wwwroot/SignIn/elements/signin-element.html (hunk 5 above)
Comment (miyconst) [inline src/SignIn/wwwroot/SignIn/elements/signin-element.html]: Done.

File: src/SignIn/Api/MainHandlers.cs (hunk 1 above)
Comment (miyconst) [inline src/SignIn/Api/MainHandlers.cs]: Done.

**Comment by warpech** [comment]
LGTM! 

Notice: SignIn, UserAdmin and Registration **develop** branches can't be merged to master before Simplified is released in Starcounter stable.

But, we can still build app versions for the warehouse by adding a `-develop.0` suffix to the version number

**Pull request merged**